pac4j-core/src/main/java/org/pac4j/core/context/ContextHelper.java

```diff
@@ -19,15 +20,15 @@
      * @param name the name of the cookie
      * @return the cookie
      */
-    public static Cookie getCookie(final Collection<Cookie> cookies, final String name) {
+    public static Optional<Cookie> getCookie(final Collection<Cookie> cookies, final String name) {
         if (cookies != null) {
```

Comment:
Surely this can be done better using streaming
cookies.stream.filter(cookie != null && CommonHelper.equals...).findFirst()

This will return an Optional directly, or empty.